src/frontend/src/AppWrapper.tsx

```diff
@@ -19,8 +28,24 @@ const queryClient = new QueryClient({
 
 export default function AppWrapper() {
     return (
-        <QueryClientProvider client={queryClient}>
-            <App />
-        </QueryClientProvider>
+        <Router>
+            <ScrollToTop />
+            <QueryClientProvider client={queryClient}>
+                <RootDarkModeProvider>
+                    <App />
+                </RootDarkModeProvider>
+            </QueryClientProvider>
+        </Router>
     );
 }
+
+// react-router sometimes doesn't scrolls to top after page-change
+export function ScrollToTop() {
+    const { pathname } = useLocation();
```

Comment:
Kann der code auch in den AppWrapper? Also über dem return ();?
Ist ja zum Glück recht wenig code.